dev-requirements.txt

```diff
@@ -0,0 +1,4 @@
+# requirements for dev
+flake8
+pylint
```

Comment:
I haven't run these checks or thought about customizing which warnings are enabled—but regardless of the settings in `setup.cfg`, doesn't having two linters seem redundant?

Reply:
Ah! Flake8 is a "if it doesn't pass, fix it", whereas pylint is more "here are some conventions you might want to check". For instance, pylint can give you a list of TODOs left in the code, or tell you about "hey, this could be a f-string", and it can be configured to ignore more or less these rules. At the end, it gives you a score you can look at to know where you stand.

And I do tend to have a perfect score on my projects... 😁 

Note that I didn't add any CI here as to not push that too far.

Reply:
Ah, there's a nuance to it! Maybe we'll reach the point of adding `pylint` upstream in Sopel itself one day. There are undoubtedly code smells it would point out that aren't covered by our existing checks like CodeQL. But this is an entirely separate discussion!

> or tell you about "hey, this could be a f-string"

Bet I can guess the first rule you always configure `pylint` to ignore… 😁

Reply:
I'll see to add tests & CI in a future PR. :)